semgrep/semgrep/semgrep_main.py

```diff
@@ -277,21 +315,32 @@ def main(
             f"running {len(all_rules)} rules from {len(valid_configs)} config{plural} {config_id_if_single} {invalid_msg}"
         )
 
+        notify_user_of_work(all_rules, include, include_dir, exclude, exclude_dir)
+
         if len(valid_configs) == 0:
             raise SemgrepError(
                 f"no valid configuration file found ({len(invalid_configs)} configs were invalid)",
                 code=MISSING_CONFIG_EXIT_CODE,
             )
 
-    # actually invoke semgrep
-    rule_matches_by_rule, debug_steps_by_rule, semgrep_errors = CoreRunner(
-        allow_exec=dangerously_allow_arbitrary_code_execution_from_rules,
-        jobs=jobs,
-        exclude=exclude,
-        include=include,
-        exclude_dir=exclude_dir,
-        include_dir=include_dir,
-    ).invoke_semgrep(targets, all_rules)
+    # Start a progress writer. This will print dots until .stop() is called.
```

Comment:
Can drop this comment since we're no longer using dots, and the class/function names are descriptive enough we probably don't need the comments anyway.